technic/machines/HV/quarry.lua

```diff
@@ -225,6 +227,39 @@ local function get_dig_pos(quarry_pos, quarry_dir, dig_pos, dig_index, dig_steps
 	return dig_pos, dig_index
 end
 
+local function should_generate_particles(pos)
+	local id = technic.pos2network(pos)
+	local net = id and technic.networks[id]
+	if net then
+		return net.lag < 35000
+	end
+	return false
+end
+
+local function dig_particles(quarry_pos, dig_pos, dig_node)
+	local t = 0.5
+	local vec = {
+		x = quarry_pos.x - dig_pos.x,
+		y = quarry_pos.y - dig_pos.y,
+		z = quarry_pos.z - dig_pos.z}
+	local mag = (vec.x^2 + vec.y^2 + vec.z^2) ^ 0.5
+	mag = (mag - 1) / mag
+	vec = {x = vec.x * mag / t, y = vec.y * mag / t, z = vec.z * mag / t}
```

Comment:
These calculations could probably be done simpler (and faster) using the `vector.*` functions: [lua_api.txt#L3325-L3376](https://github.com/minetest/minetest/blob/97248c695730c29ca91fb7e56456d061f77744f5/doc/lua_api.txt#L3325-L3376)

Reply:
Commit pushed.